mathics/core/symbols.py

```diff
@@ -333,56 +333,73 @@ def replace_slots(self, slots, evaluation) -> "Atom":
 
 class Symbol(Atom, NumericOperators, EvalMixin):
     """
-    Note: Symbol is right now used in a couple of ways which in the
-    future may be separated.
+    A Symbol is a kind of Atom that acts as a symbolic variable.
 
-    A Symbol is a kind of Atom that acts as a symbolic variable or
-    symbolic constant.
+    All Symbols have a name that can be converted to string.
 
-    All Symbols have a name that can be converted to string form.
-
-    Inside a session, a Symbol can be associated with a ``Definition``
+    A Variable Symbol is a ``Symbol``` that is associated with a ``Definition``
```

Comment:
Maybe we should also mention that a Variable is a symbol whose definition has a Downvalue.

Reply:
For example, ```System`Plus``` does not act as a variable, because it does not have an Ownvalue. As it has a Downvalue, it behaves as a "function".

Reply:
Ok - in ce9a20fe I made an attempt to mention OwnValue for variables and DownValue for functions. Feel free to suggest further changes.  It feels to me though that more detail about variable symbols and function symbols should go somewhere else that is more specific to each of these aspects.